src/engine.rs

```diff
@@ -251,37 +251,77 @@ impl Engine {
         origin: Address,
         value: U256,
         input: &[u8],
-    ) -> (ExitReason, Address) {
+    ) -> (ExitReason, SubmitResult) {
         let mut executor = self.make_executor();
         let address = executor.create_address(CreateScheme::Legacy { caller: origin });
         let (status, result) = (
             executor.transact_create(origin, value, Vec::from(input), u64::MAX),
             address,
         );
+        let used_gas = executor.used_gas();
         let (values, logs) = executor.into_state().deconstruct();
-        self.apply(values, logs, true);
-        (status, result)
+        self.apply(values, Vec::<Log>::new(), true);
+
+        let mut res_logs = Vec::new();
+        for log in logs {
+            res_logs.push(log.into());
+        }
+
+        let (res_status, used_gas) = match status {
+            ExitReason::Succeed(_) => (true, used_gas),
+            ExitReason::Revert(_) => (false, used_gas),
+            ExitReason::Error(_) | ExitReason::Fatal(_) => (false, 0),
+        };
+
+        let res = SubmitResult {
+            status: res_status,
+            gas_used: used_gas,
+            logs: res_logs,
+            result: result.0.to_vec(),
+        };
+        (status, res)
     }
 
-    pub fn call_with_args(&mut self, args: FunctionCallArgs) -> (ExitReason, Vec<u8>) {
+    pub fn submit_with_args(&mut self, args: FunctionCallArgs) -> (ExitReason, SubmitResult) {
```

Comment:
I don't think I have a strong opinion either way, but I'm just curious, what is better about the name `submit` vs `call`?

Reply:
@artob mentioned that the name `call` is wrong, and technically `submit` is a better choice of words. I do agree with that.

Reply:
@joshuajbouw That concerned the external interface, not this function. I do think this rename here is a little dubious, I don't really see how this is `submit` other than that you're now returning that result type.

@birchmd In short, `submit` is the name of the method that takes and executes a raw, signed transaction.

Reply:
Fixed in 3658d4f.